### PR TITLE
samples: bluetooth: peripheral_hids_mouse: Include SMP when security is enabled

### PR DESCRIPTION
**Pull request by maje-emb**
Refactor the sample so that BT_SMP is included when BT_HIDS_SECURITY_ENABLED is enabled.
When BT_HIDS_SECURITY_ENABLED=n, some systems disconnect from HID devices because they do not support security.

### CODE REVIEW

File: samples/bluetooth/peripheral_hids_mouse/src/main.c [hunk 1]
```diff
@@ -678,6 +681,7 @@ void button_changed(uint32_t button_state, uint32_t has_changed)
 
 	memset(&pos, 0, sizeof(struct mouse_pos));
 
+#if defined(CONFIG_BT_HIDS_SECURITY_ENABLED)
```
Comment (KAGA164) [inline samples/bluetooth/peripheral_hids_mouse/src/main.c:684]: You can you here IS_ENABLED() macro and then there won't be need to have conditional compilation in 121 line

File: samples/bluetooth/peripheral_hids_mouse/src/main.c [hunk 2]
```diff
@@ -785,8 +790,10 @@ void main(void)
 	hid_init();
 
 	k_work_init(&hids_work, mouse_handler);
-	k_work_init(&pairing_work, pairing_process);
 	k_work_init(&adv_work, advertising_process);
+#if defined(CONFIG_BT_HIDS_SECURITY_ENABLED)
```
Comment (KAGA164) [inline samples/bluetooth/peripheral_hids_mouse/src/main.c:794]: `IS ENABLED()`?

**Review by KAGA164 — COMMENTED** [review]
Maybe it would be good to add changelog entry also?

**Comment by maje-emb** [comment]
> Maybe it would be good to add changelog entry also?

Added changelog entry.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -192,6 +192,10 @@ Bluetooth samples
 
   * The Bluetooth 3-wire coex sample because of the removal of the 3-wire implementation.
 
+* :ref:`peripheral_hids_mouse` sample:
+
+  * :kconfig:option:`CONFIG_BT_SMP` is included when `CONFIG_BT_HIDS_SECURITY_ENABLED` is enabled.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:197]:
```suggestion
  * The :kconfig:option:`CONFIG_BT_SMP` Kconfig option is included when ``CONFIG_BT_HIDS_SECURITY_ENABLED`` is enabled.
```
Is that latter also a Kconfig option? At least it is not introduced in the Kconfig documentation or in the sample doc. Sounds a bit weird as something called ... _ENABLED is enabled. Should we rather say that it is selected in the Kconfig file or something like that?

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]

**Pull request merged**